x-pack/plugins/ml/public/components/field_type_icon/field_type_icon.js

```diff
@@ -11,86 +11,57 @@ import { EuiToolTip } from '@elastic/eui';
 
 // don't use something like plugins/ml/../common
 // because it won't work with the jest tests
-import { ML_JOB_FIELD_TYPES } from '../../../common/constants/field_types';
-import { FormattedMessage, injectI18n } from '@kbn/i18n/react';
+import { getMLJobTypeAriaLabel } from '../../util/field_types_utils';
+import { i18n } from '@kbn/i18n';
 
-export const FieldTypeIcon = injectI18n(function FieldTypeIcon({ tooltipEnabled = false, type, intl }) {
-  let ariaLabel = '';
-  let iconClass = '';
+export const FieldTypeIcon = ({ tooltipEnabled = false, type, needsAria = true }) => {
+  const ariaLabel = getMLJobTypeAriaLabel(type);
+
+  if (!ariaLabel) {
```

Comment:
nit, as we know that `getMLJobTypeAriaLabel` will return `null` when it can't find a label, this check should be explicitly checking for `null`

Reply:
@jgowdyelastic , good point, thanks!